Brand new project fails to compile due to incorrect types

## Steps to reproduce

```
$ npx react-native init MyApp --template react-native-template-typescript
...
$ cd MyApp/
$ npx tsc --skipLibCheck
App.tsx:12:15 - error TS2305: Module '"./node_modules/@types/react"' has no exported member 'Node'.

12  import type {Node} from 'react';
                 ~~~~

App.tsx:31:20 - error TS7031: Binding element 'children' implicitly has an 'any' type.

31  const Section = ({children, title}): Node => {
                      ~~~~~~~~

App.tsx:31:30 - error TS7031: Binding element 'title' implicitly has an 'any' type.

31  const Section = ({children, title}): Node => {
                                ~~~~~


Found 3 errors.
```

## Expected results

A newly created project should be valid Typescript (according to the accompanying `tsconfig.json`).  It would also be nice to be styled e.g. according to some prettier rules.

## Observed results

Running the app works (`npm run ios`) but compiling the code fails and VScode shows the following errors:

- React does not have a type `Node`
- `Section` component uses implicit `any` types (disallowed by strict mode configured in `tsconfig.json`)

These are both introduced in #193.

Also App.tsx shows up bright red/yellow in VScode with prettier for example because *every code line*  is indented by one space.

It would be great if the code validity could be (automatically) tested prior to releases.

--- a/template/App.tsx
+++ b/template/App.tsx
@@ -27,13 +27,9 @@
    ReloadInstructions,
  } from 'react-native/Libraries/NewAppScreen';
 
- const Section = ({
-   children,
-   title,
- }: {
-   children: React.ReactNode;
+ const Section: React.FC<{
    title: string;
- }) => {
+ }> = ({children, title}) => {
    const isDarkMode = useColorScheme() === 'dark';
    return (
      <View style={styles.sectionContainer}>

--- a/template/App.tsx
+++ b/template/App.tsx
@@ -9,7 +9,6 @@
  */
 
  import React from 'react';
- import type {Node} from 'react';
  import {
    SafeAreaView,
    ScrollView,
@@ -28,7 +27,13 @@
    ReloadInstructions,
  } from 'react-native/Libraries/NewAppScreen';
 
- const Section = ({children, title}): Node => {
+ const Section = ({
+   children,
+   title,
+ }: {
+   children: React.ReactNode;
+   title: string;
+ }) => {
    const isDarkMode = useColorScheme() === 'dark';
    return (
      <View style={styles.sectionContainer}>
@@ -54,7 +59,7 @@
    );
  };
 
- const App: () => Node = () => {
+ const App = () => {
    const isDarkMode = useColorScheme() === 'dark';
 
    const backgroundStyle = {

